File: frontend/src/User/Matches.ts
export class Matches {
  id: number | undefined;
  winner: string = "";
  loser: string = "";
  matchDate: string | undefined;

  constructor(initializer?: any) {
    if (!initializer) return;
    if (initializer.id) this.id = initializer.id;
    if (initializer.winner) this.winner = initializer.winner;
    if (initializer.loser) this.loser = initializer.loser;
    if (initializer.matchDate) this.matchDate = initializer.matchDate;
  }
}

File: src/main/java/com/capstone/pingpong/entity/Matches.java
package com.capstone.pingpong.entity;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "matches")
public class Matches {
//    @ManyToMany(mappedBy = "matchWinner")
//    Set<Players> playerWinner;
//
//    @ManyToMany(mappedBy = "matchLoser")
//    Set<Players> playerLoser;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "loser_id", referencedColumnName = "id")
    private Players loser;

    @ManyToOne
    @JoinColumn(name = "winner_id", referencedColumnName = "id")
    private Players winner;

    @Column(name = "date")
    private LocalDateTime matchDate;

    public Matches() {

    }

    public Matches(Players loser, Players winner) {
        this.loser = loser;
        this.winner = winner;
    }

    public Players getLoser() {
        return loser;
    }

    public Players getWinner() {
        return winner;
    }

//    @PostPersist
//    public void updateWinnerWinsCount() {
//        if (winner != null) {
//            winner.incrementWins();
//        }
//    }
//
//    @PostPersist
//    public void updateLooserLosesCount() {
//        if (winner != null) {
//            winner.incrementWins();
//        }
//    }
}
